### PR TITLE
Run as admin if not already

### PR DESCRIPTION
**Pull request by psyirius**
## Type of Change
- [x] UX Enhancement

## Description
Run the script as admin if detected running as a non-admin

## Impact
Will prevent the hassle of running the terminal or pwsh as admin before-hand

## Additional Information
Updated the one liner to scoped (to prevent scope pollution) and with args,
but the old one-line for the stable release will still work

## Important ⚠️
Before merging this, you should do the following
- sign and deploy the `scripts/launcher.ps1` to `https://christitus.com/win`
- remove the windev deployment from `https://christitus.com/windev`
  - `scripts/windev.ps1` is now merged into the launcher script
- sign and deploy the release `winutil.ps1` to `https://christitus.com/win/stable`
- Change the stable download url `https://christitus.com/win` to `https://christitus.com/win/stable` in `scripts/launcher.ps1` after deployment

## FAQ
- Why change `dev` to `preview` ?
  - the  `dev` version doesn't pull from the main or dev branches, it just pull a pre-release or release it seems.
  - Dev is supposed to be the tip of the tree and aliasing it with a pre-release will cause confusions

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

**Comment by MyDrift-user** [comment]
this was done something like that before, but it got changed due to antiviruses flagging it more often as a virus.
Idk how it behaves now that we have code signing..

**Comment by psyirius** [comment]
I think it will work good if got codesignature

File: scripts/start.ps1 [hunk 1]
```diff
@@ -11,6 +11,31 @@ param (
     [switch]$Run
 )
 
+# RunAs admin if not already
```
Comment (MyDrift-user) [inline scripts/start.ps1:14]: how about doing it like following example?
I've reworked layouting a bit to match the rest of the utility and made it use ps7 if possible.

```
# Check if the current user is an administrator
if (-not ([Security.Principal.WindowsPrincipal]([Security.Principal.WindowsIdentity]::GetCurrent())).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)) {
    
    # Determine which PowerShell executable to use
    if (Get-Command "pwsh" -ErrorAction SilentlyContinue) {
        $shell = "pwsh.exe"
    } else {
        $shell = "powershell.exe"
    }

    # Elevate the script and exit the current non-elevated script
    $PwshArgList = @(
        "-NoLogo",                                  # Don't print PowerShell header in CLI
        "-NoProfile",                               # Don't load PowerShell profile
        "-File", $MyInvocation.MyCommand.Source,    # Script path
        $args | ForEach-Object { $_ }               # Script arguments
    ) | ForEach-Object { "`"$_`"" }

    Start-Process -FilePath $shell -ArgumentList $PwshArgList -Verb RunAs
    exit
}
```

**Review by MyDrift-user — CHANGES_REQUESTED** [review]
Hey @psyirius 
It looks good, added the option to use Powershell 7 if possible in the review

**Review by MyDrift-user — APPROVED** [review]
nice, thanks @psyirius

**Comment by psyirius** [comment]
Looks like there's an issue.

DONT MERGE!

**Comment by psyirius** [comment]
> Looks like there's an issue.
> 
> Crashing on already admin powershell.

Works now..
Weird! 

Looks like this caused by some weird edge case occurs around the UI spawning area!

**Comment by ChrisTitusTech** [comment]
I think we can add this back in as well. Sometime AV still flags it which makes me think the elevation doesn't even matter. I'm starting to become skeptical if AV's actually do any scanning or code analyzation of that sort.

**Comment by MyDrift-user** [comment]
Hey @psyirius 

Just got an Idea of #2543, how about readding the "needs to start as admin" notification if the user presses no when it asks to run the shell as admin? idk how easy it is to track that down tho..

**Comment by psyirius** [comment]
Hmm.. We that'll be good.

Maybe process exit code?

**Comment by psyirius** [comment]
Also gotta think about the iex execution situation

**Review by MyDrift-user — APPROVED** [review]
looks good and works, thanks @psyirius

**Comment by psyirius** [comment]
> looks good and works, thanks @psyirius

Have you checked the InvokeScript Scenario?

I don't think using `iex` populates the `$MyInvocation.MyCommand.Source` with anything!

**Comment by MyDrift-user** [comment]
Did not, but will.
Before it wasn't this variable but just an irm christitus.com/win | iex.
Idk how good that now would be that we habe a dev branch..

**Comment by psyirius** [comment]
Either we use a stub or get dirty with process tokens and privileges!

**Comment by psyirius** [comment]
I'm working on a stub that might help.

**Review by MyDrift-user — CHANGES_REQUESTED** [review]
Well everything works and I generally approve exept for the `$MyInvocation.MyCommand.Source`, before there was just another call to the github url, but because we have a dev branch now I would rather try fixing this. I also tried some stuff that came to my mind but nothing worked yet xD

**Comment by psyirius** [comment]
IG a stub which then downloads and executes the main script is the painless way!

**Comment by psyirius** [comment]
@ChrisTitusTech DONT MERGE THIS NOW!

BREAKS THE ONE-LINER LAUNCHER!!!

**Review by ChrisTitusTech — APPROVED** [review]

**Comment by psyirius** [comment]
@ChrisTitusTech DONT DO IT!

**Comment by psyirius** [comment]
READ THE ABOVE COMMENTS FOR MORE

**Comment by MyDrift-user** [comment]
$myinvocation is the issue

**Comment by psyirius** [comment]
😮‍💨

**Comment by ChrisTitusTech** [comment]
Sooo I got bored and figured I would do a rewrite to fix the problems I see. 

I wanted a simple approach, but also have it use Windows terminal as it just looks a LOT better. I put in logic for wt, pwsh, and powershell depending on the system. 

Check out #2565

**Comment by ChrisTitusTech** [comment]
> @ChrisTitusTech @MyDrift-user Is dropping the script to the disk a good idea?
> 
> Kanda test the behavior of it with AVs since we currently running fileless and some AVs miss it!

I'm indifferent about that. Dropping it to $env.temp wouldn't be a big deal. Its also rather small so downloading twice isn't a problem imo.

**Comment by MyDrift-user** [comment]
> I'm indifferent about that. Dropping it to $env.temp wouldn't be a big deal. Its also rather small so downloading twice isn't a problem imo. 

The "big" difference zo last time is that we now have a dev brach, it would be cool if the functionality would support both release types.

**Comment by psyirius** [comment]
@ChrisTitusTech @MyDrift-user Read the PR description before doing anything.

**Comment by psyirius** [comment]
We can merge the windev script to this new launcher script as well. Lmk if that sounds good!

**Comment by Real-MullaC** [comment]
Hello, Chris has done this in a PR #2565 what is a much better way of doing it. As it does not require adding, changing or removing of any URL's that people are used to.

**Comment by psyirius** [comment]
> Hello, Chris has done this in a PR #2565 what is a much better way of doing it. As it does not require adding, changing or removing of any URL's that people are used to.

Read the above comment by @MyDrift-user

**Comment by Real-MullaC** [comment]
@MyDrift-user and @psyirius Chris's PR will work fine. As it gets the latest pre release what has the winutil.ps1 file in it. What then runs on the system as the pre release. So Chris's PR will work fine for this.

**Comment by psyirius** [comment]
> @MyDrift-user and @psyirius Chris's PR will work fine. As it gets the latest pre release what has the winutil.ps1 file in it. What then runs on the system as the pre release. So Chris's PR will work fine for this.

You're not getting the point here IG.
Have you checked https://github.com/ChrisTitusTech/winutil/pull/2565 from as windev and manual build (dev)

**Comment by MyDrift-user** [comment]
I think u both kind of don't get it (not trying to be mean here)

**Comment by psyirius** [comment]
> I think u both kind of don't get it (not trying to be mean here)

So, what are you suggesting?

**Pull request closed without merging**